projects/packages/publicize/src/social-image-generator/class-setup.php

```diff
@@ -73,22 +73,22 @@ public function generate_token_on_save( $post_id, $post, $update ) {
 			return;
 		}
 
-		if ( ! $publicize->has_social_image_generator_feature() ) {
-			return;
-		}
+		$settings = new Settings();
 
-		if ( wp_is_post_autosave( $post ) || wp_is_post_revision( $post_id ) ) {
+		if ( ! $settings->is_available() ) {
 			return;
 		}
 
-		if ( ! ( new Settings() )->is_enabled() ) {
```

Comment:
This prevents us from generating a token if the toggle on the admin page is off, so it needs to be removed.